speech-to-text/v1.js

```diff
@@ -371,4 +373,664 @@ SpeechToTextV1.prototype.createRecognizeStream = function(params) {
+ *
+ * Example Result:
+ * { corpora:
+   [ { out_of_vocabulary_words: 0,
+       total_words: 233,
+       name: 'test_corpus_1',
+       status: 'analyzed' },
+     { out_of_vocabulary_words: 0,
+       total_words: 0,
+       name: 'test_corpus_2',
+       status: 'being_processed' } ] }
+ *
+ * @param {Object} params
+ * @param {String} params.customization_id
+ * @param {Function} callback
+ */
+SpeechToTextV1.prototype.getCorpora = function(params, callback) {
+  var parameters = {
+    requiredParams: ['customization_id'],
+    options: {
+      method: 'GET',
+      url: '/v1/customizations/{customization_id}/corpora',
+      path: pick(params, ['customization_id']),
+      json: true
+    },
+    defaultOptions: this._options
+  };
+  return requestFactory(parameters, callback);
+};
+
+/**
+ * Delete a corpus
+ *
+ * @param {Object} params
+ * @param {String} params.customization_id
+ * @params {String} params.name - The name of the corpus that is to be deleted
+ * @param {Function} callback
+ */
+SpeechToTextV1.prototype.deleteCorpus = function(params, callback) {
+  var parameters = {
+    requiredParams: ['customization_id', 'name'],
+    options: {
+      method: 'DELETE',
+      url: '/v1/customizations/{customization_id}/corpora/{name}',
+      path: pick(params, ['customization_id', 'name']),
+      json: true
+    },
+    defaultOptions: this._options
+  };
+  return requestFactory(parameters, callback);
+};
+
+
+SpeechToTextV1.ERR_NO_CORPORA = 'ERR_NO_CORPORA';
+/**
+ * Waits while a customization status is 'pending' or 'training', fires callback once the status is 'ready' or 'available'
+ *
+ * Note: the customization will remain in 'pending' status until at least one corpus is added. Calling this on a customization with no corpa will result in an error.
+ *
+ * See http://www.ibm.com/watson/developercloud/speech-to-text/api/v1/#list_models for status details
+ *
+ * @param {Object} params
+ * @param {String} params.customization_id
+ * @param {Number} [params.interval=1000] - (milliseconds) - how log to wait between status checks
+ * @param {Number} [params.times=150] - maximum number of attempts
+ */
+SpeechToTextV1.prototype.whenCustomizationReady = function(params, callback) {
+  var self = this;
+
+  async.parallel([
+
+    // validate that it has at least one corpus
+    function(next) {
+      self.getCorpora(params, function(err, res) {
+        if (err) {
+          return next(err);
+        }
+        if (!res.corpora.length) {
+          err = new Error('Customization has no corpa and therefore will never be ready.');
+          err.code = SpeechToTextV1.ERR_NO_CORPORA;
+          return next(err)
+        }
+        next();
+      })
+    },
+
+    // check the customization status repeatedly until it's ready or avaliable
+    function(next) {
+      var options = extend({
+        interval: 1000,
```

Comment:
@kognate: It does allow interval and times to be overridden. 

@german: I originally wrote this for the integration tests, where 1 second makes a little more sense (because it's uploading tiny files), but I think you're right that the general case should probably be higher.